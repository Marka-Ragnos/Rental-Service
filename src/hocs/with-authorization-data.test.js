import React from "react";
import renderer from "react-test-renderer";
import * as Type from '../prop-types';
import withAuthorizationData from "./with-authorization-data";

const testing = () => {};

const MockComponent = (props) => {
  const {children} = props;

  return (
    <>
      {children}
    </>
  );
};

MockComponent.propTypes = {
  children: Type.MOCK_COMPONENT.isRequired,
};

const MockComponentWrapped = withAuthorizationData(MockComponent);

describe(`withAuthorizationData render correctly`, () => {
  it(`withAuthorizationData render correctly with authorization`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        email={`frontend@yandex.ru`}
        activeCity={`Paris`}
        password={123}
        onEmailChange={testing}
        onPasswordChange={testing}
        onLoginPageSubmit ={testing} >
        <React.Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`withAuthorizationData render correctly without authorization`, () => {
    const tree = renderer.create((
      <MockComponentWrapped
        activeCity={`Paris`}
        onEmailChange={testing}
        onPasswordChange={testing}
        onLoginPageSubmit ={testing} >
        <React.Fragment />
      </MockComponentWrapped>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
