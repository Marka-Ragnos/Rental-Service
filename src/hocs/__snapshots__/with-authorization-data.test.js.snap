// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuthorizationData render correctly withAuthorizationData render correctly with authorization 1`] = `null`;

exports[`withAuthorizationData render correctly withAuthorizationData render correctly without authorization 1`] = `null`;
